gcloud_bigtable/row.py

```diff
@@ -416,12 +419,13 @@ def _commit_mutate(self, timeout_seconds=None):
             mutations=mutations_list,
         )
         timeout_seconds = timeout_seconds or self.timeout_seconds
-        response = self.client.data_stub.MutateRow.async(request_pb,
-                                                         timeout_seconds)
         # We expect a `._generated.empty_pb2.Empty`.
-        response.result()
+        if async:
+            self.client.data_stub.MutateRow.async(request_pb, timeout_seconds).result()
```

Comment:
I can't recall why I did this, but **I think** it had something to do with the lack of support for error messages. Or maybe in the case of a failed request (e.g. trying to use more than 30 characters in a cluster name) the sync methods would take the entirety of the timeout seconds to report the error while the `async` methods would return altogether?

I can't quite remember.

The reason I asked @nathanielmanistaatgoogle to take a look was to get general advice on when to use one over the other.